packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
-  // we have to prepend DOCTYPE to serve valid HTML
-  ctx.body = DOCTYPE + renderToString(
-    <HTML>
-      <MyApp />
-    </HTML>
-  );
+The main entry point of this library is primarily concerned with giving you the ability to [serialize and deserialize information](https://github.com/Shopify/web-foundation/blob/master/Proposals/02%20-%20Serialization%20in%20application%20code.md) from within your application code. There are three parts to this: a `Manager`, which stores and looks up serializations, a `Provider`, that distributes this manager throughout the application, and a function for generating type-safe serialization/ deserialization components.
 
-  await next();
+#### `Manager` and `Provider`
+
+These pieces allow the library to maintain the state of your HTML document so that it can be reconstructed later on. On the server, you should construct a manager and pass it into a provider, as you will need it to construct your `<Html />` component later. You can completely omit these for the client; a version that works within the browser will automatically be constructed if you do not supply your manager/ provider.
+
+```tsx
+import {Manager, Provider} from '@shopify/react-html';
+
+function App({htmlManager}: {htmlManager: Manager}) {
+  return <Provider manager={htmlManager}>Hello world!</Provider>;
+}
+
+export default function middleware() {
+  const manager = new Manager();
+  const app = <App htmlManager={manager} />;
 }
 ```
 
-Due to [limitations in React’s implementation of HTML](https://github.com/facebook/react/issues/1035), you still need to prepend the `<!DOCTYPE html>` directive. To assist with this the module also exports a `DOCTYPE` constant.
+#### `createSerializer()`
 
-The component will automatically propagate any usage of the `react-helmet` module in your app’s content to manipulate the title or other top level HTML or HEAD attributes.
+This function creates serialize/ deserialize components. These can be used by components that manage a stateful part of the application that must be communicated from the server to the client (for example: async loaded translations, Redux, Apollo, etc).
 
-## Interface
+`createSerializer()` accepts a single string argument for the identifier to use; this will help you find the serialized `script` tag if you need to debug later on. It also accepts a generic type argument for the type of the data that will be serialized/ available after deserialization.
 
-```typescript
-export interface Props {
-  children?: React.ReactNode;
-  styles?: Asset[];
-  scripts?: Asset[];
-  blockingScripts?: Asset[];
-  headData?: {[id: string]: any};
-  data?: {[id: string]: any};
-  hideForInitialLoad?: boolean;
-}
+The function returns a pair of components:
 
-interface Asset {
-  path: string;
-  integrity?: string;
-}
+```tsx
+createSerializer<string>('MyData');
 
-interface Browser {
-  userAgent: string;
-  supported: boolean;
-}
+// Would create the following components:
```

Comment:
```suggestion
// Serialize and WithSerialized have the following typings:
```

I'm not sure if typings is the right word there :/

Reply:
Perhaps this should be in two codeblocks to reenforce the idea that these functions are the result of what you just called, instead of being something a user copy+pastes